crud/len.lua

```diff
@@ -0,0 +1,72 @@
+local checks = require('checks')
+local errors = require('errors')
+local vshard = require('vshard')
+
+local call = require('crud.common.call')
+local dev_checks = require('crud.common.dev_checks')
+
+local LenError = errors.new_class('LenError',  {capture_stack = false})
+
+local len = {}
+
+local LEN_FUNC_NAME = '_crud.len_on_storage'
+
+local function len_on_storage(space_name)
+    dev_checks('string')
+
+    local space = box.space[space_name]
+    if space == nil then
+        return nil, LenError:new("Space %q doesn't exist", space_name)
+    end
+
+    return space:len()
+end
+
+function len.init()
+    _G._crud.len_on_storage = len_on_storage
+end
+
+--- Calculates the number of tuples in the space for memtx engine
+--- Calculates the maximum approximate number of tuples in the space for vinyl engine
+--
+-- @function call
+--
+-- @param string space_name
+--  A space name
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout
+--
+-- @return[1] number
+-- @treturn[2] nil
+-- @treturn[2] table Error description
+--
+function len.call(space_name, opts)
+    checks('string', {
+        timeout = '?number',
+    })
+
+    opts = opts or {}
+
+    local replicasets = vshard.router.routeall()
+    local results, err = call.map(LEN_FUNC_NAME, {space_name}, {
+        mode = 'write',
+        replicasets = replicasets,
+        timeout = opts.timeout,
+    })
```

Comment:
You can use `vshard.rouer.map_callrw` (https://github.com/tarantool/vshard/commit/a394e3f4008a7436f011c81028dcaadc270eebf6) for such purpose. It gives more guarantees in case when rebalanser is active.